domain/domain.py

```diff
@@ -97,44 +93,40 @@ def setup_attributes(self):
         self.total = total
         self.single_stats = single_stats
         tic = time.clock()
-        self.pair_stats = self.preproc_pair_stats(pair_stats)
+        self.pair_stats = self._topk_pair_stats(pair_stats)
         toc = time.clock()
         prep_time = toc - tic
         logging.debug("DONE with pair stats preparation in %.2f secs", prep_time)
         self.setup_complete = True
 
-    def preproc_pair_stats(self, pair_stats):
+    def _topk_pair_stats(self, pair_stats):
         """
-        preproc_pair_stats converts 'pair_stats' which is a dictionary mapping
-            { attr1 -> { attr2 -> DataFrame } } where
+        _topk_pair_stats converts 'pair_stats' which is a dictionary mapping
+            { attr1 -> { attr2 -> {val1 -> {val2 -> count } } } } where
                 DataFrame contains 3 columns:
-                  <attr1>: all possible values for attr1 ('val1')
-                  <attr2>: all values for attr2 that appeared at least once with <val1> ('val2')
-                  <count>: frequency (# of entities) where attr1: val1 AND attr2: val2
+                  <val1>: all possible values for attr1
+                  <val2>: all values for attr2 that appeared at least once with <val1>
+                  <count>: frequency (# of entities) where attr1: <val1> AND attr2: <val2>
 
         to a flattened 4-level dictionary { attr1 -> { attr2 -> { val1 -> [Top K list of val2] } } }
         i.e. maps to the Top K co-occurring values for attr2 for a given
         attr1-val1 pair.
         """
 
         out = {}
-        for key1 in tqdm(pair_stats):
-            if key1 == '_tid_':
-                continue
-            out[key1] = {}
-            for key2 in pair_stats[key1]:
-                if key1 == '_tid_':
-                    continue
-                df = pair_stats[key1][key2]
-                if not df.empty:
-                    out[key1][key2] = dictify(df)
-                    for val in out[key1][key2]:
-                        denominator = self.single_stats[key1][val]
-                        tau = float(self.topk*denominator)
-                        top_cands = [k for (k, v) in out[key1][key2][val].items() if v > tau]
-                        out[key1][key2][val] = top_cands
-                else:
-                    out[key1][key2] = {}
+        for attr1 in tqdm(pair_stats.keys()):
+            out[attr1] = {}
+            for attr2 in pair_stats[attr1].keys():
+                out[attr1][attr2] = {}
+                for val1 in pair_stats[attr1][attr2].keys():
+                    try:
+                        denominator = self.single_stats[attr1][val1]
+                    except KeyError:
+                        import pdb; pdb.set_trace()
+
+                    tau = float(self.topk*denominator)
+                    top_cands = [val2 for (val2, count) in pair_stats[attr1][attr2][val1].items() if count > tau]
```

Comment:
is `count` an accurate term for the scores?

Reply:
Yes, `count` is the co-occurrence frequency between `val1` and `val2` in `attr1` and `attr2`, respectively. `topk` is actually a misnomer: it is in fact the `topXpercent` of pairwise co-occurrence frequencies (actually: this is exactly true, I've added a TODO here since I don't think we're computing this as we intended).

Reply:
thanks for adding docs about it